### DISCUSSION TITLE
NFT activity

### DISCUSSION THREAD

**Discussion by labralords** [njaremko/iota-lib-rs#1688] [Q&A] [answered]
Hi,

I have a couple of questions really...

- Is there a way to retrieve all NFT related activity on the shimmer network?
- Can I retrieve activity related to a NFT collection? 
- Can I retrieve activity related to a NFT?
- Is there a way to list all NFT collection ids available?

Thanks in advance

**Comment by Thoralf-M** [comment, marked as answer]
Regarding the first three questions: since the core nodes prune data and the inx-indexer only keeps track of unspent, you can only keep track of the NFT activity "live" via MQTT or get the UTXO changes for every milestone as long as they aren't pruned, but this way you can't get historic activity unless you use another inx plugin like https://github.com/iotaledger/inx-chronicle which keeps the full history (this has an own API for the explorer, but that's not supported by iota.rs)

For the last question you can get all current NFT outputs with `client.nft_output_ids([])`, `client.get_outputs(&output_ids_response.items)` and then you have to look into the immutable metadata of each NFT output to know whether it's part of a NFT collection or not and create a HashSet with the ids

**Reply by labralords** [reply]
Thanks for the quick response! I'll try out some of these methods to see what suits my project the best.

So is there a way to get hold of pruned data at all? Will the inx-chronicle be able to fetch pruned data?

**Reply by Thoralf-M** [reply]
You would need to start inx-chronicle with the data from genesis, there is no way to get it from somewhere else (unless you also have a node that never prunes data from the genesis)

**Reply by labralords** [reply]
Thanks! Maybe obvious, but where can I get hold of data from genesis?

**Reply by Thoralf-M** [reply]
This "golden db" has all all data since genesis https://files.shimmer.shimmer.network/?prefix=dbs/golden-db/
